### PR TITLE
Added job selection interface

### LINKED ISSUE #24: Display jobs on CLI and let user select one of them
Fetch jobs that the current user responsible for.
Display them on CLI.

### PR DESCRIPTION
**Pull request by ilayda21**
## Done
- Jobs are retrieved.
- "interactive" argument is added
- Required argument validation is added.

## QA
- Checkout the feature branch
- Install dependencies with `yarn`
- Update "Job.ts" > "clonePost" method > boardPost variable :
`const boardToPost = boards.find((board) => board.name === "Test Board");`
- Run `ts-node ./src/index.ts add-job -i`
- Verify no required argument error is thrown.
- Verify jobs that has "Recruiter" tag are displayed.
- Verify only one job can be selected.
- Verify the selected job ID's is logged.
- Run `ts-node ./src/index.ts add-job`
- Verify an error that indicates some arguments missing is thrown.
- Run `ts-node ./src/index.ts delete-jobs -i`
- Verify no required argument error is thrown.
- Verify jobs that has "Recruiter" tag are displayed.
- Verify only one job can be selected.
- Verify the selected job ID's is logged.
- Run `ts-node ./src/index.ts delete-jobs`
- Verify an error that indicates some arguments missing is thrown.

## Issue
Fixes  #24

### CODE REVIEW

**Comment by tbille** [comment]
Have 0 jobs where you are a hiring lead, run the command and you get a non human error: https://pastebin.canonical.com/p/MpxJHcDh5R/

File: src/common/pageUtils.ts [hunk 1]
```diff
@@ -59,3 +59,14 @@ export async function sendRequest(
...
+
+export async function getIDFromURL(
+    element: Puppeteer.ElementHandle,
+    selector: string
+): Promise<number> {
+    const url = await element.$eval(selector, (anchor: Element) =>
+        anchor.getAttribute("href")
+    );
+    const urlParts: string[] = ("" + url).split("/");
+    return +urlParts[urlParts.length - 1];
```
Comment (goulinkh) [inline src/common/pageUtils.ts:71]: This should be `parseInt`

File: src/index.ts [hunk 2]
```diff
@@ -1,46 +1,122 @@
...
+import { red } from "colors";
+import ora from "ora";
+// @ts-ignore This can be deleted after https://github.com/enquirer/enquirer/issues/135 is fixed.
+import { Select } from "enquirer";
 
-async function addPosts(jobID: number, regions: string[], cloneFrom: number) {
-    const sso = new SSO();
-    const loginCookies = await sso.login();
-    console.log(green("✓"), "Authentication complete");
+async function getJobFromCLI(job: Job, message: string, spinner: ora.Ora) {
```
Comment (goulinkh) [inline src/index.ts:11]: I think it's better to name it how it's getting the job:
something like `getJobInteractive`

File: src/index.ts [hunk 3]
```diff
@@ -1,46 +1,122 @@
...
+
+    try {
+        if (isInteractive) {
+            const id = await getJobFromCLI(
+                job,
+                "What job would you like to create job posts for?",
+                spinner
+            );
+            console.log(`Job ID: ${id}`);
+            // TODO add posts
```
Comment (goulinkh) [inline src/index.ts:45]: I think that you forgot to remove these lines

File: src/index.ts [hunk 4]
```diff
@@ -1,46 +1,122 @@
...
+        } else {
+            if (!jobID) throw Error(`Job ID argument is missing.`);
+            if (!regions) throw Error(`Region parameter is missing.`);
+            const jobData: JobInfo = await job.getJobData(jobID);
```
Comment (goulinkh) [inline src/index.ts:49]: before this line you can update the status message of the spinner, something like `Getting the job data`

File: src/index.ts [hunk 5]
```diff
@@ -1,46 +1,122 @@
...
 
-    // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
-    await job.clonePost(jobData.posts, regions, cloneFrom);
+            // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
+            await job.clonePost(jobData.posts, regions, cloneFrom);
```
Comment (goulinkh) [inline src/index.ts:52]: same here, update the spinner status all the time to let the user know what's happening

File: src/index.ts [hunk 6]
```diff
@@ -1,46 +1,122 @@
...
+            // Mark all newly added job posts as live
+            await job.markAsLive(jobID, jobData.posts);
+        }
+    } catch (error) {
+        spinner.stop();
```
Comment (goulinkh) [inline src/index.ts:57]: the spinner should stop in all the cases so it should be in `finally`

File: src/index.ts [hunk 7]
```diff
@@ -1,46 +1,122 @@
...
-    browser.close();
+    try {
+        if (isInteractive) {
+            const id = await getJobFromCLI(
+                job,
+                "What job would you like to delete job posts from?",
+                spinner
+            );
+            console.log(`Job ID: ${id}`);
+            // TODO delete posts
```
Comment (goulinkh) [inline src/index.ts:83]: Same here, dev notes?

File: src/index.ts [hunk 8]
```diff
@@ -1,46 +1,122 @@
...
+        } else {
+            if (!jobID) throw Error(`Job ID argument is missing.`);
+            await job.deletePosts(jobID, regions, similar);
```
Comment (goulinkh) [inline src/index.ts:86]: Same here, update the spinner message status

File: src/index.ts [hunk 9]
```diff
@@ -1,46 +1,122 @@
...
+        }
+    } catch (error) {
+        spinner.stop();
```
Comment (goulinkh) [inline src/index.ts:89]: move it to the `finally` section

**Review by goulinkh — CHANGES_REQUESTED** [review]

File: src/automations/Job.ts [hunk 10]
```diff
@@ -198,4 +198,55 @@ export default class Job {
         if (!jobAnchor) throw new Error("Cannot found the job");
```
Comment (goulinkh) [inline src/automations/Job.ts:198]: I just noticed this error in the message

**Review by goulinkh — CHANGES_REQUESTED** [review]

File: src/common/pageUtils.ts (hunk 1 above)
Comment (ilayda21) [reply src/common/pageUtils.ts:71]: :+1:

File: src/index.ts (hunk 2 above)
Comment (ilayda21) [reply src/index.ts:11]: :+1:

File: src/index.ts (hunk 3 above)
Comment (ilayda21) [reply src/index.ts:49]: Console log is added for QA.

File: src/index.ts (hunk 4 above)
Comment (ilayda21) [reply src/index.ts:53]: :+1:

File: src/index.ts (hunk 5 above)
Comment (ilayda21) [reply src/index.ts:56]: :+1:

File: src/index.ts (hunk 6 above)
Comment (ilayda21) [reply src/index.ts:61]: :+1:

File: src/index.ts (hunk 7 above)
Comment (ilayda21) [reply src/index.ts:87]: :+1:

File: src/index.ts (hunk 8 above)
Comment (ilayda21) [reply src/index.ts:90]: :+1:

File: src/index.ts (hunk 9 above)
Comment (ilayda21) [reply src/index.ts:93]: :+1:

File: src/index.ts [hunk 11]
```diff
@@ -1,40 +1,109 @@
...
+// @ts-ignore This can be deleted after https://github.com/enquirer/enquirer/issues/135 is fixed.
+import { Select } from "enquirer";
+
+async function getJobInteractive(job: Job, message: string, spinner: ora.Ora) {
+    // display UI
+    const jobs = await job.getJobs();
+    spinner.succeed();
+    if (jobs.size === 0)
+        throw Error(
+            `Only hiring leads can create job posts, make sure you are one or contact HR.`
```
Comment (tbille) [inline src/index.ts:17]: Small update from @marinacastejon 

```suggestion
            `Only hiring leads can create job posts. If you are not sure about your hiring role please contact HR.`
```

File: src/common/pageUtils.ts [hunk 12]
```diff
@@ -59,3 +59,14 @@ export async function sendRequest(
...
 }
+
+export async function getIDFromURL(
+    element: Puppeteer.ElementHandle,
+    selector: string
+): Promise<number> {
+    const url = await element.$eval(selector, (anchor: Element) =>
+        anchor.getAttribute("href")
+    );
+    const urlParts: string[] = ("" + url).split("/");
```
Comment (goulinkh) [inline src/common/pageUtils.ts:70]: I'm not sure what's `""+url` for?

File: src/index.ts (hunk 11 above)
Comment (ilayda21) [reply src/index.ts]: :+1:

**Review by tbille — APPROVED** [review]
QA+1
Tested:

- no hiring leads jobs
- one job hiring lead

It worked as expected.

File: src/common/pageUtils.ts (hunk 12 above)
Comment (goulinkh) [reply src/common/pageUtils.ts:70]: if it's to cast to string you can do so with something more TS friendly like: `string(url)`

**Review by goulinkh — APPROVED** [review]
LGTM ⚡

File: src/common/pageUtils.ts (hunk 12 above)
Comment (ilayda21) [reply src/common/pageUtils.ts]: instead i added a null check

**Pull request merged**